### PR TITLE
Added Timeout to ament_uncrustify

### PR DESCRIPTION
**Pull request by ahcorde**
There are many timeout related with uncrustify on [Windows builds](https://ci.ros2.org/view/nightly/job/nightly_win_deb/3054/consoleText)

This PR adds the TIMEOUT argument to ament_uncrustify and increase the default timeout for Windows up to 300

### CODE REVIEW

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake [hunk 1]
```diff
@@ -28,6 +28,8 @@
 # :param MAX_LINE_LENGTH: override the maximum line length,
 #   the default is defined in ament_uncrustify
 # :type MAX_LINE_LENGTH: integer
+# :param TIMEOUT: the test timeout in seconds, default: 300
```
Comment (clalancette) [inline ament_cmake_uncrustify/cmake/ament_uncrustify.cmake:31]:
```suggestion
# :param TIMEOUT: the test timeout in seconds, default (Windows): 300, default (other): 60
```

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake [hunk 2]
```diff
@@ -65,6 +67,17 @@ function(ament_uncrustify)
     list(APPEND cmd "--exclude" "${ARG_EXCLUDE}")
   endif()
   list(APPEND cmd ${ARG_UNPARSED_ARGUMENTS})
+  if(NOT ARG_TIMEOUT)
+    if(WIN32)
```
Comment (clalancette) [inline ament_cmake_uncrustify/cmake/ament_uncrustify.cmake:71]: I think it is worth putting a comment here as to why we are increasing the Windows timeout.

**Review by clalancette — CHANGES_REQUESTED** [review]
Two small changes to documentation, but otherwise this looks good.

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI.

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20798/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15101/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21476/)

**Pull request merged**